src/frontend/src/common/analytics/eventTypes.ts

```diff
@@ -82,13 +85,19 @@ export enum EVENT_TYPES {
  * that kicks off the underlying Segment `track` by ensuring that all events
  * extend from that base structure.
  */
-type EventValue = string | number | boolean | null;
+type EventValue = string | number | boolean | FilterDate | null;
 export type EventData = Record<string, EventValue | undefined>;
 
 // While we only send values of EventValue, sometimes we send a JSON string.
 // For ease of readability below, we alias string for those cases.
 type JsonString = string;
 
+// For sending date types back to segment (e.g. for filtering on upload or collection dates)
```

Comment:
Following up on what I said above with EventValue, we almost always want to avoid having objects be what we send as event data. Since this ultimately lands as data in DB columns for our Data Analyst teammate, we want to try to avoid sending anything other than atomic `EventValue` data. Simple, flat objects with atomic values for everything is what we want to emit. There are rarely some cases where we break this rule, and in that case we fall back to JSONifying objects before smushing them into event data payloads. For some examples of this, look for references elsewhere in the codebase to the above `JsonString` type.